mesonbuild/modules/pkgconfig.py

```diff
@@ -120,8 +120,15 @@ def _process_libs(self, libs, public):
                 processed_reqs.append(obj.generated_pc)
             elif isinstance(obj, dependencies.PkgConfigDependency):
                 if obj.found():
-                    processed_reqs.append(obj.name)
-                    self.add_version_reqs(obj.name, obj.version_reqs)
+                    if obj.static:
+                        # in case of embedded static dependency
+                        for arg in obj.link_args:
+                            if not os.sep in arg:
+                                # pull external dependencies
+                                processed_reqs.append(arg)
```

Comment:
Since it's a link_arg, it should go to processed_libs, I think. processed_reqs is for pkg-config names.